### PR TITLE
[TKW] e2e transpose

### PR DESCRIPTION
**Pull request by Hardcode84**

### CODE REVIEW

File: tests/kernel/wave/wave_e2e_test.py [hunk 1]
```diff
@@ -60,3 +60,117 @@ def test(
...
+    ]
+    constraints += [tkw.WorkgroupConstraint(M, BLOCK_M, 1)]
+    constraints += [tkw.WorkgroupConstraint(N, BLOCK_N, 0)]
+    constraints += [tkw.WaveConstraint(M, BLOCK_M)]
+    constraints += [tkw.WaveConstraint(N, BLOCK_N)]
+
+    i = tkw.IndexMapping.iterator(0)
+    j = tkw.IndexMapping.iterator(1)
+    mapping = tkw.IndexMapping(
+        num_iterators=2, inputs={M: i, N: j}, outputs={M: i, N: j}
```
Comment (raikonenfnu) [inline tests/kernel/wave/wave_e2e_test.py:89]: NIT: Can we make `inputs={N: i, M: j}, outputs={N: i, M: j}` and set `a: tkl.Memory[M, N, ..` and `b: tkl.Memory[N, M...` to make the map more obviously transpose

**Review by raikonenfnu — COMMENTED** [review]
Overall looks good, just 1 nit :)

File: tests/kernel/wave/wave_e2e_test.py (hunk 1 above)
Comment (Hardcode84) [reply tests/kernel/wave/wave_e2e_test.py]: done

**Review by raikonenfnu — APPROVED** [review]
LGTM!

**Pull request merged**